navigator_api/api/main.py

```diff
@@ -93,79 +96,55 @@ def workflow_state(dataset_id):
     workflow = get_or_create_workflow(dataset['id'], current_user.id, name=dataset['title'])
     if not workflow:
         return error.error_response(500, f"Couldn't get decision engine details for dataset {dataset_id}")
-
+    engine_decision = engine_client.get_decision(ckan_cli, dataset_id, skip_actions=workflow.skipped_tasks)
     return jsonify({
         "id": f"{workflow.id}",
-        "milestones": [
-            {
-                "id": "xxx",
-                "title": "Naomi Data Prep",
-                "completed": True,
-                "progress": 100
-            },
-            {
-                "id": "yyy",
-                "title": "Shiny 90 Data Prep",
-                "completed": False,
-                "progress": 50
-            },
-            {
-                "id": "zzz",
-                "title": "Update Spectrum",
-                "completed": False,
-                "progress": 0
-            }
-        ],
-        "taskBreadcrumps": [
-            "aaabbbbcc",
-            "dddeeefff",
-            "asdfasdfa",
-            "aaabbbbcc",
-            "dddeeefff",
-            "asdfasdfa",
-            "aaabbbbcc",
-            "dddeeefff",
-            "asdfasdfa"
-        ],
+        "progress": engine_decision["progress"]["progress"],
+        "milestones": engine_decision["progress"]["milestones"],
+        "milestoneListFullyResolved": engine_decision["progress"]["milestoneListFullyResolved"],
+        "taskBreadcrumps": engine_decision["actions"],
```

Comment:
Typo I think? "breadcrumps"

Reply:
the spellings is correct I think. It's intentionally called "taskBreadcrumps" according to ui & api design.